js/lib/ui.river.js

```diff
@@ -1,27 +1,12 @@
+/**
+ * @deprecated 2.2
+ */
 elgg.provide('elgg.ui.river');
 
 elgg.ui.river.init = function() {
-	$(document).on('change', '#elgg-river-selector', function() {
-		var url = window.location.href;
-		if (window.location.search.length) {
-			url = url.substring(0, url.indexOf('?'));
-		}
-		url += '?' + $(this).val();
-		elgg.forward(url);
-	});
-
-	$(document).on('elgg_ui_toggle', function (e, data) {
-		var $toggle = $(e.target);
-		var $elements = data.$toggled_elements;
-
-		if ($elements.is('.elgg-river-responses > .elgg-form-comment-save')) {
-			if ($toggle.hasClass('elgg-state-active')) {
-				$elements.find('.elgg-input-text').focus();
-			} else {
-				$toggle.blur();
-			}
-		}
-	});
+	elgg.deprecated_notice('ui.river.js library has been deprecated. Update core/river/filter and forms/comment/save views to require component AMD modules instead.', '2.2')
+	require(['core/river/filter']);
```

Comment:
This is registered as a JS and not loaded by default.
It's loaded by the selector view, so the form binding lives on pure luck.
Deprecation is there juat in case somebody calls elgg_load_js ()
On Mar 22, 2016 1:57 PM, "Steve Clay" <notifications@github.com> wrote:

> In js/lib/ui.river.js
> <https://github.com/Elgg/Elgg/pull/9536#discussion_r56980769>:
>
> > -	});
> > -
> > -	$(document).on('elgg_ui_toggle', function (e, data) {
> > -		var $toggle = $(e.target);
> > -		var $elements = data.$toggled_elements;
> > -
> > -		if ($elements.is('.elgg-river-responses > .elgg-form-comment-save')) {
> > -			if ($toggle.hasClass('elgg-state-active')) {
> > -				$elements.find('.elgg-input-text').focus();
> > -			} else {
> > -				$toggle.blur();
> > -			}
> > -		}
> > -	});
> > +	elgg.deprecated_notice('ui.river.js library has been deprecated. Update core/river/filter and forms/comment/save views to require component AMD modules instead.', '2.2')
> > +	require(['core/river/filter']);
>
> Put these requires in an anon function bound to init, system. Then a
> stock install won't get the deprecation notice.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9536/files/89c64354c58fca53a8ba044d74c7d68c1a8f4cd8#r56980769>
>